Solar-Wrapper/src/TrackablesSolARImpl.cpp

```diff
@@ -0,0 +1,287 @@
+
+    void TrackablesSolARImpl::get_trackables(Pistache::Http::ResponseWriter &response)
+    {
+
+        //initialize the json object that we will send back to the client
+        auto jsonObjects = nlohmann::json::array();
+
+        //declaration of all the objects that will be changed at each iteration of the loop
+        nlohmann::json toAdd;
+        Trackable track;
+
+        std::vector<SRef<datastructure::StorageTrackable>> vector;
+        if(m_worldStorage->getTrackables(vector) != FrameworkReturnCode::_SUCCESS)
+        {
+            //Exception raised in getTrackables
+            response.headers().add<Pistache::Http::Header::ContentType>(MIME(Text, Plain));
+            response.send(Pistache::Http::Code::Internal_Server_Error, "Something went wrong when fetching the world storage");
+        }
+        else
+        {
+            //iteration over the content of the world storage
+            for (const SRef<StorageTrackable> &t : vector){
+                //add the current trackable to the JSON object
+                track = fromStorage(*t);
+                to_json(toAdd, track);
+                jsonObjects.push_back(toAdd);
+            }
+
+            //send the JSON object to the client
+            response.headers().add<Pistache::Http::Header::ContentType>(MIME(Application, Json));
+            response.send(Pistache::Http::Code::Ok, jsonObjects.dump());
+        }
+    }
+
+    Trackable TrackablesSolARImpl::fromStorage(StorageTrackable trackable){
+        //the object to be returned
+        Trackable ret;
+
+        //convert all the StorageTrackable attributes into Trackable attibutes
+
+        //trackable UUID
+        std::string id = xpcf::uuids::to_string(trackable.getID());
+        ret.setUUID(id);
+
+        //creator UUID
+        std::string creatorUid = xpcf::uuids::to_string(trackable.getCreatorID());
+        ret.setCreatorUUID(creatorUid);
+
+        //Trackable type
+        std::string type = resolveTrackableType(trackable.getType());
+        ret.setTrackableType(type);
+
+        //EncodingInfo struct
+        EncodingInformationStructure encodingInfo;
+        EncodingInfo storageEncodingInfo(trackable.getEncodingInfo());
+        encodingInfo.setDataFormat(storageEncodingInfo.getDataFormat());
+        encodingInfo.setVersion(storageEncodingInfo.getVersion());
+        ret.setTrackableEncodingInformation(encodingInfo);
+
+        //Payload
+        std::vector<std::byte> storagePayload = trackable.getPayload();
+        std::string payload(reinterpret_cast<const char *>(&storagePayload[0]), storagePayload.size());
+        ret.setTrackablePayload(payload);
+
+        //Transform3Df (localCRS)
+        Transform3Df transform3d = trackable.getLocalCrs();
+        std::vector<float> localCRS;
+        size_t nCols = transform3d.cols();
+        for (size_t i = 0; i < nCols; ++i)
+           for (size_t j = 0; j < nCols; ++j)
+           {
+                localCRS.push_back(transform3d(j, i));
```

Comment:
Are you sure about this loops ? I'm not thoroughly checking this, but two things looks fishy:
* both loops are using `cols` as limit (It might work here, Transform3D being a 4x4 matrix I guess, so nCols=nRows), but it would be more legible if you used nRows for either variable so we know the intentions better
* Matrix are row major, so shouldn't it be `transform3d(i, j)` ? (just asking)